src/WebApiContrib.Core.Versioning/Strategies/IVersionStrategy.cs

```diff
@@ -15,6 +15,6 @@ public interface IVersionStrategy
         /// <param name="context">The <see cref="HttpContext"/> of the current request.</param>
         /// <param name="routeData">The <see cref="RouteData"/>of the current request.</param>
         /// <returns>A version integer, or <c>null</c> if a version cannot be determined.</returns>
-        int? GetVersion(HttpContext context, RouteData routeData);
+        VersionContext GetVersion(HttpContext context, RouteData routeData);
```

Comment:
That's why I suggested making the return type `VersionContext?` (or `VersionResult?`) and making the `version` argument of `VersionContext` required. Does it make sense to return a result, but no `version`?